axelor-supplychain/src/main/resources/views/SaleOrderLine.xml

```diff
@@ -79,13 +79,25 @@
           <field name="supplierPartner" showIf="saleSupplySelect == 2"
             onSelect="action-sale-order-line-method-supplier-partner-domain"
             form-view="partner-form" grid-view="partner-grid"/>
-          <field name="billOfMaterial"
-            onSelect="action-sale-order-line-attrs-bill-of-material-domain"
-            if-module="axelor-production" form-view="bill-of-material-form"
-            grid-view="bill-of-material-grid" if="__config__.app.isApp('production')" canEdit="true"/>
-          <button name="customizeBOMBtn" title="Customize BOM"
-            onClick="action-production-method-customize-bill-of-material"
-            if-module="axelor-production" if="__config__.app.isApp('production')"/>
+          <spacer colSpan="6"/>
+          <panel name="billOfMaterialPanel" if-module="axelor-production">
```

Comment:
on top of the module check, we should also check if the production app is activated.